pkg/api/scylla/validation/scylladbcluster_validation.go

```diff
@@ -0,0 +1,450 @@
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("remoteKubernetesClusterName"), dc.RemoteKubernetesClusterName, msg))
+	}
+
+	return allErrs
+}
+
+func ValidateScyllaDBClusterDatacenterTemplate(dcTemplate *scyllav1alpha1.ScyllaDBClusterDatacenterTemplate, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if dcTemplate.RackTemplate != nil {
+		allErrs = append(allErrs, ValidateScyllaDBDatacenterRackTemplate(dcTemplate.RackTemplate, fldPath.Child("rackTemplate"))...)
+	}
+
+	allErrs = append(allErrs, validateStructSliceFieldUniqueness(dcTemplate.Racks, func(rack scyllav1alpha1.RackSpec) string {
+		return rack.Name
+	}, "name", fldPath.Child("racks"))...)
+
+	for rackIdx, rack := range dcTemplate.Racks {
+		if len(rack.Name) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("racks").Index(rackIdx).Child("name"), "rack name must not be empty"))
+		}
+		allErrs = append(allErrs, ValidateScyllaDBDatacenterRackTemplate(&rack.RackTemplate, fldPath.Child("racks").Index(rackIdx))...)
+	}
+
+	if dcTemplate.TopologyLabelSelector != nil {
+		allErrs = append(allErrs, apimachinerymetav1validation.ValidateLabels(dcTemplate.TopologyLabelSelector, fldPath.Child("topologyLabelSelector"))...)
+	}
+
+	if dcTemplate.ScyllaDB != nil {
+		if dcTemplate.ScyllaDB.Storage != nil {
+			if dcTemplate.ScyllaDB.Storage.Metadata != nil {
+				allErrs = append(allErrs, apimachinerymetav1validation.ValidateLabels(dcTemplate.ScyllaDB.Storage.Metadata.Labels, fldPath.Child("scyllaDB", "storage", "metadata", "labels"))...)
+				allErrs = append(allErrs, apimachineryvalidation.ValidateAnnotations(dcTemplate.ScyllaDB.Storage.Metadata.Annotations, fldPath.Child("scyllaDB", "storage", "metadata", "annotations"))...)
+			}
+
+			storageCapacity, err := resource.ParseQuantity(dcTemplate.ScyllaDB.Storage.Capacity)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "storage", "capacity"), dcTemplate.ScyllaDB.Storage.Capacity, fmt.Sprintf("unable to parse capacity: %v", err)))
+			} else if storageCapacity.CmpInt64(0) <= 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "storage", "capacity"), dcTemplate.ScyllaDB.Storage.Capacity, "must be greater than zero"))
+			}
+
+			if dcTemplate.ScyllaDB.Storage.StorageClassName != nil {
+				for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(*dcTemplate.ScyllaDB.Storage.StorageClassName, false) {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "storage", "storageClassName"), *dcTemplate.ScyllaDB.Storage.StorageClassName, msg))
+				}
+			}
+		}
+
+		if dcTemplate.ScyllaDB.CustomConfigMapRef != nil {
+			for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(*dcTemplate.ScyllaDB.CustomConfigMapRef, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "customConfigMapRef"), *dcTemplate.ScyllaDB.CustomConfigMapRef, msg))
+			}
```

Comment:
Seems to duplicate with `types_scylladbdatacenter.go`. Also why isn't this split into functions?